src/argilla/client/feedback/dataset.py

```diff
@@ -549,46 +549,38 @@ def delete_and_raise_exception(dataset_id: UUID, exception: Exception) -> None:
                     datasets_api_v1.delete_dataset(client=httpx_client, id=dataset_id)
                 except Exception as e:
                     raise Exception(
-                        "Failed while deleting the `FeedbackTask` dataset with ID"
-                        f" '{dataset_id}' from Argilla with exception: {e}"
-                    )
+                        f"Failed while deleting the `FeedbackTask` dataset with ID '{dataset_id}' from Argilla with exception: {e}"
+                    ) from e
                 raise exception
 
             for field in self.fields:
                 try:
                     datasets_api_v1.add_field(client=httpx_client, id=argilla_id, field=json.loads(field.json()))
                 except Exception as e:
-                    delete_and_raise_exception(
-                        dataset_id=argilla_id,
-                        exception=Exception(
-                            f"Failed while adding the field '{field.name}' to the"
-                            f" `FeedbackTask` dataset in Argilla with exception: {e}"
-                        ),
+                    exc = Exception(
+                        f"Failed while adding the field '{field.name}' to the `FeedbackTask` dataset in Argilla with exception: {e}"
                     )
+                    exc.__cause__ = e
+                    delete_and_raise_exception(dataset_id=argilla_id, exception=exc)
 
             for question in self.questions:
                 try:
                     datasets_api_v1.add_question(
                         client=httpx_client, id=argilla_id, question=json.loads(question.json())
                     )
                 except Exception as e:
-                    delete_and_raise_exception(
-                        dataset_id=argilla_id,
-                        exception=Exception(
-                            f"Failed while adding the question '{question.name}' to the"
-                            f" `FeedbackTask` dataset in Argilla with exception: {e}"
-                        ),
+                    exc = Exception(
+                        f"Failed while adding the question '{question.name}' to the `FeedbackTask` dataset in Argilla with exception: {e}"
                     )
+                    exc.__cause__ = e
+                    delete_and_raise_exception(dataset_id=argilla_id, exception=exc)
 
             try:
                 datasets_api_v1.publish_dataset(client=httpx_client, id=argilla_id)
             except Exception as e:
-                delete_and_raise_exception(
-                    dataset_id=argilla_id,
-                    exception=Exception(
-                        f"Failed while publishing the `FeedbackTask` dataset in Argilla with exception: {e}"
-                    ),
-                )
+                exc = Exception(f"Failed while publishing the `FeedbackTask` dataset in Argilla with exception: {e}")
+                exc.__cause__ = e
```

Comment:
maybe we should use the conventional way to do this? https://docs.python.org/3/library/exceptions.html#exception-context
```python
delete_dataset(dataset_id=argilla_id)
raise exc from e
```

Reply:
Yes, I think that's better than doing weird stuff like this :) So `delete_and_raise_exception` -> `delete_dataset`, right?

Reply:
exactly!